### PR TITLE
Update lightning version to v1.2

### LINKED ISSUE #132: align with PR 1.2.0
## 🚀 Feature

Atm, we have a hard link to 1.2.0rc0 which is a bit outdated...
Probably some API adjust need to be done, continue in #62

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes #132 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by teddykoker — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-pytorch-lightning==1.2.0rc0  # todo: we shall align with real 1.2
+pytorch-lightning==1.2.0
```
Comment (Borda) [inline requirements.txt:1]:
```suggestion
pytorch-lightning>=1.2.0
```

**Comment by Borda** [comment]
@carmocca @kaushikb11 can we continue the `predict` talk here 🐰

**Comment by Borda** [comment]
@kaushikb11 any progress here? it would be nice to have Flash release working fine with mainstream PL (not the hack RC we have now) cc: @aribornstein

**Comment by kaushikb11** [comment]
@Borda Let me look at the failing tests today, weren't failing before.

**Comment by kaushikb11** [comment]
@Borda Had a talk with Carlos. We can't point to lightning 1.2 unless #141 PR is in, and there are other changes required in Lightning as well.

**Comment by carmocca** [comment]
Also #130. And updating the Lightning Trainer to call `model.predict_step` instead of `model.predict`
cc: @justusschock

**Review by edgarriba — COMMENTED** [review]
IDK about the finetunning functions but the change of requirements is needed to unbock other branches like datapipeline. If not inconvenient I'd merge. /cc @kaushikb11 @tchaton

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@kaushikb11 there are some PL related error, such as misconfiguration, mind have a look

**Comment by kaushikb11** [comment]
@Borda Sure!

**Comment by kaushikb11** [comment]
One Blocking PR https://github.com/PyTorchLightning/pytorch-lightning/pull/6541

**Comment by Borda** [comment]
> One Unblocking PR [PyTorchLightning/pytorch-lightning#6541](https://github.com/PyTorchLightning/pytorch-lightning/pull/6541)

which means we would need to sen min version to 1.2.4 or higher...

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-pytorch-lightning==1.2.0rc0  # todo: we shall align with real 1.2
+pytorch-lightning>=1.2.0
```
Comment (carmocca) [inline requirements.txt:1]:
```suggestion
pytorch-lightning>=1.2.5
```

**Pull request merged**